README.md

```diff
@@ -1,2 +1,61 @@
-# tc-build
-A set of scripts to build LLVM and binutils
+# Toolchain build scripts
+
+There are times where a tip of tree LLVM build will have some issue fixed and it isn't available to you, maybe because it isn't in a release or it isn't available through your distribution's package management system. At that point, to get that fix, LLVM needs to be compiled, which sounds scary but is [rather simple](https://llvm.org/docs/GettingStarted.html). The `build-llvm.py` script takes it a step farther by trying to optimize both LLVM's build time by:
+
+* Trimming down a lot of things that kernel developers don't care about:
+  * Documentation
+  * LLVM tests
+  * Ocaml bindings
+  * libfuzzer
+* Building with the faster tools available (in order of fastest to slowest):
+  * clang + lld
+  * clang/gcc + ld.gold
+  * clang/gcc + ld.bfd
+
+It also aims to make kernel compilation go a little bit faster by generating a toolchain that is optimized for the host CPU (using `-O2 -march=native -mtune=native`).
+
+## Getting started
+
+These scripts have been tested in a Docker image of the following distributions, with the following packages installed:
+
+* ### Debian/Ubuntu
+
+  ```
+  apt install ca-certificates ccache clang cmake curl file gcc g++ git make ninja-build python3 texinfo zlib1g-dev
+  ```
+
+  On Debian Buster or Ubuntu Bionic/Cosmic/Disco, `apt install lld` should be added as well for faster compiles.
+
+* ### Fedora
+
+  ```
+  dnf install ccache clang cmake gcc gcc-c++ git lld make ninja-build python3 zlib-devel
+  ```
+
+* ### Arch Linux
+
+  ```
+  pacman -S base-devel ccache clang cmake git lld ninja python3
+  ```
+
+Python 3.5.3+ is recommended, as that is what the script has been tested against. These scripts should be distribution agnostic. Please feel free to add different distribution install commands here through a pull request.
+
+## build-llvm.py
+
+By default, `./build-llvm.py` will clone LLVM, grab the latest binutils tarball (for the LLVMgold.so plugin), and build LLVM, clang, and lld, and install them into `./usr`. Run `./build-llvm.py -h` for more options.
+
+## build-binutils.py
+
+This script builds a standalone copy of binutils. By default, `./build-binutils.py` will download binutils 2.32, build for all architectures we currently care about (arm32, aarch64, powerpc32, powerpc64le, and x86_64), and install them into `./usr`. Run `./build-binutils.py -h` for more options.
```

Comment:
`./usr`, just confirming, not top level `/usr`, but `./usr`?

Reply:
Can it be called `build` if it's a local dir? It's much more common for open source projects to recommend creating a build directory so that you're not polluting the sources with object files, and `build` is pretty standard, IME.

Reply:
> `./usr`, just confirming, not top level `/usr`, but `./usr`?

Correct so that it is like `/usr` but self contained and local (and doesn't require root).

> Can it be called `build` if it's a local dir? It's much more common for open source projects to recommend creating a build directory so that you're not polluting the sources with object files, and `build` is pretty standard, IME.

Well, the script builds the files in `build/<repo>` then installs them to `usr`. Are you saying you would rather the files be installed to `build`? I'd like LLVM and binutils to be installed into the same folder so that only one `PATH` modification is needed, rather than two.

Reply:
e019b68c608ce38890820dbb1a55c02ddb870869